src/wp-includes/rest-api.php

```diff
@@ -1814,6 +1876,54 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 		}
 	}
 
+	if ( isset( $args['anyOf'] ) ) {
+		$is_valid_found = false;
```

Comment:
Let's extract these to a separate function after we see that the schema contains the `anyOf` keyword.

Reply:
Something we could do instead, is to adjust the error code from `rest_invalid_param` to `rest_invalid_type` in `rest_validate_value_from_schema` when the value fails validation because of the type. Then we don't need to repeat the validation logic here and can instead filter out errors based on that error code.